apps/api/src/bank-transactions-file/bank-transactions-file.controller.ts

```diff
@@ -0,0 +1,130 @@
+import { parseBankTransactionsFile } from './helpers/parser'
+
+@Controller('bank-transactions-file')
+export class BankTransactionsFileController {
+  constructor(
+    private readonly bankTransactionsFileService: BankTransactionsFileService,
+    @Inject(forwardRef(() => VaultService)) private vaultService: VaultService,
+    private readonly donationsService: DonationsService,
+    private readonly campaignService: CampaignService,
+    @Inject(forwardRef(() => PersonService)) private readonly personService: PersonService,
+  ) {}
+
+  @Post(':bank_transactions_file_id')
+  @UseInterceptors(FilesInterceptor('file'))
+  async create(
+    @Param('bank_transactions_file_id') bankTransactionsFileId: string,
+    @Body() body: FilesTypesDto,
+    @UploadedFiles() files: Express.Multer.File[],
+    @AuthenticatedUser() user: KeycloakTokenParsed,
+  ) {
+    const keycloakId = user.sub as string
+    const person = await this.personService.findOneByKeycloakId(keycloakId)
+    if (!person) {
+      Logger.warn('No person record with keycloak ID: ' + keycloakId)
+      throw new NotFoundException('No person record with keycloak ID: ' + keycloakId)
+    }
+
+    //checking for file size
+
+    files.forEach((file) => {
+      if (file.size > 1048576) {
+        Logger.error('File bigger than 1MB' + file.filename)
+        throw new PayloadTooLargeException('File bigger than 1MB' + file.filename)
+      }
+    })
+
+    const allMovements: { payment: CreateBankPaymentDto; paymentRef: string }[][] = []
+    let accountMovementsForAFile: { payment: CreateBankPaymentDto; paymentRef: string }[] = []
+
+    const promises = await Promise.all(
+      files.map((file, key) => {
+        accountMovementsForAFile = parseBankTransactionsFile(file.buffer)
+        allMovements.push(accountMovementsForAFile)
+        const filesType = body.types
+        return this.bankTransactionsFileService.create(
+          Array.isArray(filesType) ? filesType[key] : filesType,
+          file.originalname,
+          file.mimetype,
+          bankTransactionsFileId,
+          person,
+          file.buffer,
+        )
+      }),
+    )
+    for (const fileOfBankMovements of allMovements) {
+      for await (const movement of fileOfBankMovements) {
+        const campaign = await this.campaignService.getCampaignByPaymentReference(
+          movement.paymentRef,
+        )
+        if (!campaign) {
+          Logger.warn('No campaign with payment reference: ' + movement.paymentRef)
+          throw new NotFoundException('No person record with keycloak ID: ' + keycloakId)
+        }
+        const vault = await this.vaultService.findByCampaignId(campaign.id)
+        movement.payment.extPaymentMethodId = 'imported file bank payment'
+        movement.payment.targetVaultId = vault[0].id
+        movement.payment.personsEmail = person.email
+        const donation = await this.donationsService.createBankPayment(movement.payment)
```

Comment:
Sending SQL queries in a loop can cause harm to the db availability. Please consider using `prisma.donations.createMany()`  and send all records with a single (internally batched) query.